### PR TITLE
should not ignore `options.transport` function provided in `Sentry.init(...)`

### PR DESCRIPTION
**Pull request by zivl**
## :loudspeaker: Type of change

- [X] Bugfix
- [ ] New feature
- [ ] Enhancement
- [ ] Refactoring

## :scroll: Description
I've updated the ReactNativeClient constructor to first check if `options.transport` is provided and if so - then it takes it and calls to `super(options)`. If `options.transport` not provided then keep the previous logic and conduct react-native transport layer and then calls to `super(options)`.

## :bulb: Motivation and Context
this resolves issue #2397 

## :green_heart: How did you test it?
I've added a test which gets the given custom transport layer and then runs and validates the customized `send` and `flush` functions.

## :pencil: Checklist

- [x] I reviewed submitted code
- [x] I added tests to verify changes
- [x] All tests passing
- [x] No breaking changes

### CODE REVIEW

File: test/client.test.ts
Comment (liamjones) [inline test/client.test.ts:117]: If you pulled the Transport send/flush functions to variables you could have an even stronger check of: `expect(client.getTransport()?.flush).toBe(myFlushFunction)`

Comment (zivl) [reply test/client.test.ts:117]: yeah I guess that's another option, shall I change it or it is not critical?

Comment (liamjones) [reply test/client.test.ts:117]: 🤷 I defer to the Sentry maintainers on that one 😄

Comment (marandaneto) [reply test/client.test.ts:117]: Yep I'd prefer the stronger check too.

**Comment by marandaneto** [comment]
@zivl please add the changelog entry.
Thank you for fixing this.

File: test/client.test.ts
Comment (zivl) [reply test/client.test.ts]: done

**Comment by zivl** [comment]
@marandaneto change requests are done. please review again

**Comment by marandaneto** [comment]
> @marandaneto change requests are done. please review again

Awesome.
Double-check the linter.

```
/home/runner/work/sentry-react-native/sentry-react-native/test/client.test.ts
  118:14  error  Avoid referencing unbound methods which may cause unintentional scoping of `this`  @typescript-eslint/unbound-method
  11[9](https://github.com/getsentry/sentry-react-native/runs/7539677797?check_suite_focus=true#step:5:10):14  error  Avoid referencing unbound methods which may cause unintentional scoping of `this`  @typescript-eslint/unbound-method
```

When fixing this, we are ready to LGTM.

**Comment by zivl** [comment]
@marandaneto I can't get out of it without really put a `eslint-ignore...` comment as this is a dummy implementation for the sake of the test. it is also stated [here](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/unbound-method.md#when-not-to-use-it) that it is better to just add an ignore comment, can you go with that? or have any other solution?

**Comment by marandaneto** [comment]
> @marandaneto I can't get out of it without really put a `eslint-ignore...` comment as this is a dummy implementation for the sake of the test. it is also stated [here](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/unbound-method.md?rgh-link-date=2022-07-27T14%3A41%3A58Z#when-not-to-use-it) that it is better to just add an ignore comment, can you go with that? or have any other solution?

We can ignore it, no problem.

**Review by marandaneto — APPROVED** [review]

**Comment by zivl** [comment]
@marandaneto, @bruno-garcia can you pls run workflows and merge?

**Pull request merged**
